### PR TITLE
Add popular tags in first column of profile section

### PR DESCRIPTION
**Pull request by CleverFool77**
Fixes https://github.com/publiclab/plots2/issues/5891

Screenshot from 2019-08-09 18-50-07

### CODE REVIEW

**Comment by CleverFool77** [comment]
Hi @jywarren @gautamig54 In the design we have two rows of tags below the topic cards, In this one of the tags section is of preferred tags right ? which is usually added by user to its profile through this ?

Screenshot from 2019-08-09 18-33-43

**Comment by CleverFool77** [comment]
Recent update. Just pushed these changes - 

Screenshot from 2019-08-09 18-50-07

**Comment by gautamig54** [comment]
> Hi @jywarren @gautamig54 In the design we have two rows of tags below the topic cards, In this one of the tags section is of preferred tags right ? which is usually added by user to its profile through this ?
> 
> Screenshot from 2019-08-09 18-33-43

Yes!

File: app/views/users/profile.html.erb [hunk 1]
```diff
@@ -7,8 +7,24 @@
   <%= render :partial => "map/userLeaflet" , locals: { haslocation: false, user: @profile_user } %>
   <% end %>
 
-  <hr />
-  <br />
+  <br>
```
Comment (newbazz) [inline app/views/users/profile.html.erb]: please keep up with the coding style, br should have a closing tag as it earlier had.

File: app/views/users/profile.html.erb [hunk 2]
```diff
@@ -7,8 +7,24 @@
...
+  <div style="text-align: center;">
+    <span id="tags">
+      <% tags = Tag.sort_according_to_followers(@profile_user.tags,"desc") %>
+      <% tags.each do |tag| %>
+          <span id="tag_<%= tag.id %>" style=""><a class="badge badge-primary mb-1" style="color: white;" href="/tag/<%= tag.name %>"><%= tag.name %></a></span>
+      <% end %>
+    </span>
+  </div>
+
+  <br>
```
Comment (newbazz) [inline app/views/users/profile.html.erb]: closing tag here as well.

Comment (CleverFool77) [reply app/views/users/profile.html.erb]: cool. I'll fix it.

**Comment by gautamig54** [comment]
Hi @CleverFool77 Just wanted to ask that these popular tags are specific to each user or in general popular tags of the website?

**Comment by jywarren** [comment]
Ah, @gautamig54 has a good point, this is a bit ambiguous. How is it described in the mockup? Can we clarify this via design or labeling?

**Comment by CleverFool77** [comment]
Hi @gautamig54 @jywarren This popular tags are specific to each user !!!!

**Comment by CleverFool77** [comment]
Hi @gautamig54 @jywarren This PR is for popular tag for specific user, I'll soon add another PR which would be for specific tags preferred by user and has been added by user in its profile.

**Comment by jywarren** [comment]
Ah, so it's tags commonly used on posts by that user? Maybe we could change
the text to "Most common topics"? Or something else to be a little more
clear, what do you think?

**Comment by CleverFool77** [comment]
Hi Jeff.
If this would be considered as most common topics then what would be
heading for the tags preferred by user. Is it most preferred tags?

**Comment by jywarren** [comment]
Well, is it a list of tags the user has created on their own OR others'
posts, or is it a list of tags that most commonly occur on posts by the
user? I guess we should choose one (my preference is the 2nd, as that
reflects the topical scope of their posts, rather than them helping to
categorize others' posts) but happy to hear your views!

**Comment by CleverFool77** [comment]
Hi Jeff.
Reagrding this, I'm not sure. I thought
We would have two types of tags.
First it would be tags mostly used by user which is PR right now for
popular or most common tags.
And another would be list of tags that user has added in it's profile.
For this one, we'll add a PR soon.

**Comment by jywarren** [comment]
Ah, I'm sorry! I was entirely talking about tags on notes, not user tags!

For user tags, I think we'd best just keep them quite separate from the
node tags. In the sidebar seems a good basic way to keep them separate.

**Comment by CleverFool77** [comment]
Cool. The user tags are on sidebar already. We just need to fix some bugs
and it would be okay.

Thanks!!

**Comment by jywarren** [comment]
Yeah that sounds great, it's what i understood from the screenshots, great!

**Comment by CleverFool77** [comment]
Just updated !!

Screenshot from 2019-08-16 11-30-12

**Comment by jywarren** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">There was an error with Danger bot’s Junit parsing: No JUnit file was found at output.xml</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          3 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">@CleverFool77 Thank you for your pull request! I’m here to help with some tips and recommendations. Please take a look at the list provided and help us review and accept your contribution! And <strong>don’t be discouraged if you see errors</strong> – we’re here to help.</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This pull request doesn’t link to a issue number. Please refer to the issue it fixes (if any) in the body of your PR, in the format: <code>Fixes #123</code>.</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">#<RuntimeError: No="" JUnit="" file="" was="" found="" at="" output.xml=""></RuntimeError:></td>
    </tr>
  </tbody>
</table>

<details><summary>Screenshots :camera_flash: (click to expand)</summary><h3>6120-test_viewing_question_post.png</h3><p></p><h3>6120-test_wiki.png</h3><p></p><h3>6120-test_tag_page.png</h3><p></p><h3>6120-test_login.png</h3><p></p><h3>6120-test_wiki_page_with_inline_grids.png</h3><p></p><h3>6120-test_questions.png</h3><p></p><h3>6120-test_methods.png</h3><p></p><h3>6120-test_tag_by_author_page.png</h3><p></p><h3>6120-test_viewing_the_dropdown_menu.png</h3><p></p><h3>6120-test_viewing_the_settings_page.png</h3><p></p><h3>6120-test_comments.png</h3><p></p><h3>6120-test_stats.png</h3><p></p><h3>6120-test_tags.png</h3><p></p><h3>6120-test_people.png</h3><p></p><h3>6120-test_front.png</h3><p></p><h3>6120-test_signup.png</h3><p></p><h3>6120-test_questions_shadow.png</h3><p></p><h3>6120-test_blog.png</h3><p></p><h3>6120-test_question_page.png</h3><p></p><h3>6120-test_front_page_with_navbar_search_autocomplete.png</h3><p></p><h3>6120-failures_test_blog_page_with_location_modal.png</h3><p></p><h3>6120-test_viewing_the_dashboard.png</h3><p></p></details><p><a href='https://github.com/publiclab/plots2/issues/5316'>Learn about automated screenshots</a></p>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by CleverFool77** [comment]
I'm not sure why travis is failing. Everything seemed to pass locally but not here !!

**Comment by jywarren** [comment]
Hm, can you click through into the tests and read the logs to see what the error is, and paste it back in here? Thanks!

**Comment by CleverFool77** [comment]
Hi @jywarren The error tests doesn't seem to be related to this PR. 
It's something related to blog page modal !!
```
ERROR["test_blog_page_with_location_modal", #<Minitest::Reporters::Suite:0x00007f380ef874e0 @name="ScreenshotsTest">, 156.71562014200003]
 test_blog_page_with_location_modal#ScreenshotsTest (156.72s)
Capybara::ElementNotFound:         Capybara::ElementNotFound: Unable to find css "a.blurred-location-input"
            test/system/screenshots_test.rb:100:in `block in <class:ScreenshotsTest>
```

Thanks !!
I have closed and reopened this PR again to start the build. Let's see whats the result !!

**Comment by CleverFool77** [comment]
Now all the checks seem to have passed after restarting the build.
Thanks !!

**Comment by CleverFool77** [comment]
This PR seems ready !!! @jywarren 
Thanks !!

**Comment by jywarren** [comment]
Great!!!! Super work!

**Pull request merged**